Makefile

```
@@ -79,12 +79,12 @@ $(PREFIX_PROG_STRIPPED)%.img: $(PREFIX_PROG_STRIPPED)%.elf
 	$(SIL)$(OBJCOPY) -O binary $< $@
 
 
-$(PREFIX_O)/$(TARGET_FAMILY)-$(TARGET_SUBFAMILY).ld:
+$(PREFIX_O)/$(TARGET_FAMILY)-$(TARGET_SUBFAMILY).ld: | $(OBJS)
 	@echo "GEN $(@F)"
```

Maybe just add `mkdir`
```suggestion
	@echo "GEN $(@F)"
	mkdir -p "$(@D)"
```

---

Thanks for the suggestion, but it does not change the order. It's expected that `GEN` is done before the `LD` and after `CC` objects are build. Previously `GEN` was started right after `plo` build begins (see [before](https://github.com/phoenix-rtos/plo/runs/5666535778?check_suite_focus=true#step:5:335), and with [this PR](https://github.com/phoenix-rtos/plo/runs/5680619437?check_suite_focus=true#step:5:390)). The suggestion of course fixes random `nonexistent directory` but does not fix second issue with `LD` which also manifests itself in random race condition and error. So, if `$(OBJS)` are build $`(PREFIX_O)` directory exist, then `LD` may be generated - so this PR fixes both issues.

---

Something I don't understand.
Why should the `*.ld` file be generated only after all `*.o` files have been created?
I understand that the error occurs when creating the `*.elf` file ?
What kind of error?

---

Yes when `elf` file is being created, the error manifests when using `distcc` to build all available targets in parallel, just adding `mkdir` does not help in case of `distcc`, it randomly stops at not existing e.g. `_build/armv7a9-zynq7000/plo/phfs/msg.o` or objects from plo/lib/, use of ordered-only prerequisites helps in that case.

---

From what you wrote, it seems that the problem is with `distcc` ( with `make -j` running on a single station this error does not occur).
It creates a `*.elf` file if all the files it depends on have not yet been created.
If `distcc` behaves like this, your solution probably not guaranteed to work correctly either.
However I suggest adding `mkdir` when creating the `*.ld` file.

---

Don't get me wrong. Adding `mkdir` the way you suggest is awkward for me, because it doesn't treat directory as a dependency. To be well understood, I push next change to this PR (to be squashed later), maybe it will explain more why I think so. For the moment I'm marking this discussion as resolved because it applies to the first commit, I very please discuss on review of the next commit.